Net pricing in CHM API (#43)
MewsSystems/gitbook-channel-manager-api: mews-api.md

| Property | Type |  | Description |
| --- | --- | --- | --- |

> | `amount` | `decimal` | required | Defines the amount of the absolute fee. Sent in `gross` |

Reviewer: No dot at the end of sentence.

  * The whole group is uniquely identified by `channelManagerId` in Mews and in the channel manager extranet.
  * Each reservation should have a unique code within the group. The same code for the reservation should be provided in any following modification message.
  * Each reservation in the group can have different `start`, `end`, `spaceTypeCode`, `ratePlanCode`.
* Group total cost `group`.`totalAmount` is the sum of each `reservation`.`totalAmount`, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.

>   * If for any reason the `group`.`totalAmount` is be different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.

Reviewer 2: is different

Reviewer 2: To add one more line: 
A reservation group `amount` should be either gross or net. Both `grooss` and `net` amounts, or one of them should be sent.

  * The whole group is uniquely identified by `channelManagerId` in Mews and in the channel manager extranet.
  * Each reservation should have a unique code within the group. The same code for the reservation should be provided in any following modification message.
  * Each reservation in the group can have different `start`, `end`, `spaceTypeCode`, `ratePlanCode`.

> * Group total cost `group`.`totalAmount` is the sum of each `reservation`.`totalAmount`, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.

Reviewer 3: - Replace 1st phrase with,
"`group`.`totalAmount` is the sum of all `reservation`.`totalAmount` objects in the group,".

> * Group total cost `group`.`totalAmount` is the sum of all `reservation`.`totalAmount` objects in the group, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.

Reviewer: I'd make dot part of the code fragment...

Reviewer 3: Do you mean the dots between `group`.`totalCost` and `reservation`.`totalAmount` or the dot after `reservation`?

>   * If for any reason the `group`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.

Reviewer 3: Replace with,
"If for any reason the sum of the `reservation`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved."

>   * A reservation `group` amounts should be either `gross` or `net`. Either one value can be sent. 

Author: ```suggestion
  * A reservation `group` amounts should be either `gross` or `net`. At least one value has to be sent. 
```

>   * If for any reason the sum of the `reservation`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.

Reviewer 3: ```suggestion
  * If for any reason the sum of the `reservation`.`totalAmount` objects is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.
```

>   * A reservation `group` amounts should be either `gross` or `net`. At least one value has to be sent. 

Reviewer 3: Remove "a", "reservation `group` amounts" is plural.

        {
          "code": "1",
          "cost": 20,
          "amount": 

>           {

Reviewer 4: a brace should be on the previous line

        {
          "code": "1",

>           "cost": 20,

Reviewer: Shouldn't we remove all the obsolete things from the examples and also from the documentation?

Author: @siroky I left them there until we remove them completely we have issue with deadline to remove them completely

Reviewer: I know, usually in connector, we delete from documentation immediately, but support in code until deadline. Discuss with @vitasamek

        {
          "code": "1",
          "cost": 20,
          "amount": 

>           {

Author: ```suggestion
          "amount": {
```

        120,
        120
      ],

>       "amounts": [

Reviewer: Consider using the same formatting as in Connector API - meaning 4 space indentation. Especially with very nested examples, it's much more readable IMHO.

      "to": "2020-05-09"
    }
  ],
  "totalCost": 600,
  "totalAmount": 

>   {

Reviewer 4: a brace should be on the previous line

>   {

Author: ```suggestion
  "totalAmount": {
```

| `channelId` | `string` | required \(always\) | Unique identification of the booking in the Channel \(i.e. OTA\). |
| `channelManagerId` | `string` | required \(always\) | Unique identification of the booking in the channel manager. |
| `currencyCode` | `string` | required \(exc. Cancellation\) | 3 letter code of currency of any price within the booking. |

> | ~~`totalCost`~~ | ~~`decimal`~~ | ~~required \(exc. Cancellation\)~~ | ~~Total cost of the whole booking.~~ |

Reviewer: This I'd rather fully remove, just like we do it in Connector.

| `ratePlanCode` | `string` | required \(exc. Cancellation\) | Rate type code of the reservation. |
| `from` | `string` | required \(exc. Cancellation\) | Start date in format `"yyyy-MM-dd"` \(e.g. `"2017-12-24"` for Christmas Eve\). |
| `to` | `string` | required \(exc. Cancellation\) | End date in format `"yyyy-MM-dd"` \(e.g. `"2017-12-31"` for New Years Eve\). |
| ~~`totalCost`~~ | ~~`decimal`~~ | ~~required \(exc. Cancellation\)~~ | ~~Total cost of the reservation.~~ |
| `totalAmount` | [`Amount`](mews-api.md#amount) object | required \(exc. Cancellation\) | Total amount of the reservation. |
| `adultCount` | `int` | required \(exc. Cancellation\) | Number of adults per the reservation. |
| `childCount` | `int` | optional \(exc. Cancellation\) | Number of children per the reservation. |
| `state` | `int` | optional | [Reservation State](mews-api.md#reservation-states) code of reservation state. _If not provided, Mews will handle the reservation as _`Created`_ or _`Modified`_._ |

> | `prices` | `decimal` collection | required \(exc. Cancellation\) | Collection of prices for each night of the reservation. _The count of prices in this collection has to correspond with number of nights in the reservation._ |

Reviewer 3: Isn't this depreciated?

| `amounts` | [`Amount`](mews-api.md#amount) collection | required \(exc. Cancellation\) | Collection of amounts for each night of the reservation. _The count of amounts in this collection has to correspond with number of nights in the reservation._ |

> | `extras` | [`Extra`](mews-api.md#extra) collection | optional | Collection of extra ordered products for the reservation \(e.g. Breakfast\). _Their total price is included in the _`totalCost`_ of the reservation._ |

Reviewer 3: Needs to refer to _`totalAmount`_.

#### Extra

* Total cost of the extra product should be sent in `net` or `gross` amounts. Either both `gross` and `net` amounts, or one of them should be sent.

| Property | Type |  | Description |
| --- | --- | --- | --- |
| `code` | `string` | required | Mapping code of the extra product. |
| `cost` | `decimal` | required | Total cost of the extra product. |
| `count` | `int` | required | Count of extra products ordered. |
| `pricing` | `int` | required | [`Extra pricing Type`](mews-api.md#extra-pricing-types) code of the extra product pricing. |

#### Amount

| Property | Type |  | Description |
| --- | --- | --- | --- |
| `net` | `decimal` | optional | Price with taxes excluded. |
| `gross` | `decimal` | optional | Price with taxes included. |

> 

Author: ```suggestion
```

#### Extra

> * Total cost of the extra product should be sent in `net` or `gross` amounts. Either both `gross` and `net` amounts, or one of them should be sent.

Reviewer 3: This says sending `gross`, `net`, or both are supported. Why is this different from line 568?

| `count` | `int` | required | Count of extra products ordered. |
| `pricing` | `int` | required | [`Extra pricing Type`](mews-api.md#extra-pricing-types) code of the extra product pricing. |

#### Amount

| Property | Type |  | Description |
| --- | --- | --- | --- |
| `net` | `decimal` | optional | Price with taxes excluded. |
| `gross` | `decimal` | optional | Price with taxes included. |

> 

Reviewer 4: Remove the empty line
